src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py

```diff
@@ -589,15 +590,43 @@ def test_network_nic(self, resource_group):
             JMESPathCheckV2('NewNIC.ipConfigurations[0].privateIpAllocationMethod', 'Dynamic'),
             JMESPathCheckV2('NewNIC.provisioningState', 'Succeeded')
         ])
-        # exercise optional parameters
-        self.cmd('network nic create -g {} -n {} --subnet {} --ip-forwarding --private-ip-address {} --public-ip-address {} --internal-dns-name test --dns-servers 100.1.2.3 --lb-address-pools {} --lb-inbound-nat-rules {} --accelerated-networking'.format(rg, nic, subnet_id, private_ip, public_ip_name, address_pool_ids, rule_ids), checks=[
-            JMESPathCheckV2('NewNIC.ipConfigurations[0].privateIpAllocationMethod', 'Static'),
-            JMESPathCheckV2('NewNIC.ipConfigurations[0].privateIpAddress', private_ip),
-            JMESPathCheckV2('NewNIC.enableIpForwarding', True),
-            JMESPathCheckV2('NewNIC.enableAcceleratedNetworking', True),
-            JMESPathCheckV2('NewNIC.provisioningState', 'Succeeded'),
-            JMESPathCheckV2('NewNIC.dnsSettings.internalDnsNameLabel', 'test'),
-            JMESPathCheckV2('length(NewNIC.dnsSettings.dnsServers)', 1)])
+
+        # exercise optional parameters; enableAcceleratedNetworking parameter is not available in '2017-03-09-profile'
+        if get_active_api_profile() == '2017-03-09-profile':
+            template = 'network nic create -g {} -n {} --subnet {} --ip-forwarding --private-ip-address {} --public-ip-address {} --internal-dns-name test --dns-servers 100.1.2.3 --lb-address-pools {} --lb-inbound-nat-rules {}'
+            self.cmd(template.format(rg, nic, subnet_id, private_ip, public_ip_name, address_pool_ids, rule_ids), checks=[
+                JMESPathCheckV2('NewNIC.ipConfigurations[0].privateIpAllocationMethod', 'Static'),
+                JMESPathCheckV2('NewNIC.ipConfigurations[0].privateIpAddress', private_ip),
+                JMESPathCheckV2('NewNIC.enableIpForwarding', True),
+                JMESPathCheckV2('NewNIC.provisioningState', 'Succeeded'),
+                JMESPathCheckV2('NewNIC.dnsSettings.internalDnsNameLabel', 'test'),
+                JMESPathCheckV2('length(NewNIC.dnsSettings.dnsServers)', 1)
+            ])
+            self.cmd('network nic update -g {} -n {} --internal-dns-name noodle --ip-forwarding true --dns-servers "" --network-security-group {}'.format(rg, nic, alt_nsg), checks=[
+                JMESPathCheckV2('enableIpForwarding', True),
+                JMESPathCheckV2('dnsSettings.internalDnsNameLabel', 'noodle'),
+                JMESPathCheckV2('length(dnsSettings.dnsServers)', 0),
+                JMESPathCheckV2("networkSecurityGroup.contains(id, '{}')".format(alt_nsg), True)
+            ])
+        else:
+            template = 'network nic create -g {} -n {} --subnet {} --ip-forwarding --private-ip-address {} --public-ip-address {} --internal-dns-name test --dns-servers 100.1.2.3 --lb-address-pools {} --lb-inbound-nat-rules {} --accelerated-networking'
```

Comment:
Help content doesn't display `"--accelerated-networking" `for` "2017-03-09-profile`". But verified it does for `"latest"` profile.